edx_rest_api_client/tests/test_client.py

```diff
@@ -44,6 +46,32 @@ def test_valid_configuration(self, kwargs, auth_type):
         api = EdxRestApiClient(**kwargs)
         self.assertEqual(auth_type, type(getattr(api._store["session"], "auth")))  # pylint: disable=protected-access
 
+    @httpretty.activate
+    @ddt.unpack
+    @ddt.data(
+        ({'url': URL, 'signing_key': None, 'username': None, 'oauth_access_token': None}, NoneType),
+        ({'url': URL, 'signing_key': None, 'username': None, 'oauth_access_token': None, 'append_slash': False},
+         NoneType)
+    )
+    def test_valid_configuration_with_append_slash(self, kwargs, auth_type):
+        """
+        Verify that if a url has no trailing slash then the reset api client
+        will append the url with trailing slash by default and if the option
+        'append_slash' is set to False the the url will remain unchanged.
+        """
+        api = EdxRestApiClient(**kwargs)
+        self.assertEqual(auth_type, type(getattr(api._store["session"], "auth")))  # pylint: disable=protected-access
+
+        # call the api with some invalid resource url which will result in 404
+        with self.assertRaises(exceptions.HttpNotFoundError):
+            api.invalid_resource.get()
+
+        expected_path = '/api/v2/invalid_resource/'
+        if kwargs.get('append_slash') is not None and not kwargs.get('append_slash'):
+            expected_path = '/api/v2/invalid_resource'
+
+        self.assertEqual(httpretty.last_request().path, expected_path)
+
```

Comment:
i feel like this is just testing slumber itself and doesn't belong here.  I think it would make more sense to include the no-slash check in tests that will be written for accessing the user api (or other apis that do not use ending slashes).